src/bokeh/util/paths.py

```diff
@@ -18,52 +18,53 @@
 #-----------------------------------------------------------------------------
 
 # Standard library imports
-import sys
-from os.path import (
-    abspath,
-    dirname,
-    isdir,
-    join,
-    normpath,
-    realpath,
-)
+from pathlib import Path
 
 #-----------------------------------------------------------------------------
 # Globals and constants
 #-----------------------------------------------------------------------------
 
 # Root dir of Bokeh package
-ROOT_DIR = dirname(dirname(abspath(__file__)))
+ROOT_DIR = Path(__file__).absolute().resolve().parent.parent
 
 __all__ = (
-    'serverdir',
-    'bokehjsdir',
+    "serverdir",
+    "bokehjsdir",
 )
 
 #-----------------------------------------------------------------------------
 # General API
 #-----------------------------------------------------------------------------
 
-def serverdir() -> str:
-    """ Get the location of the server subpackage
+def serverdir() -> Path:
+    """
+    Get the location of the server subpackage.
     """
-    path = join(ROOT_DIR, 'server')
-    path = normpath(path)
-    if sys.platform == 'cygwin': path = realpath(path)
-    return path
+    return ROOT_DIR / "server"
 
+def staticdir() -> Path:
```

Comment:
We will definitely want a migration note about the type change of the return. We could also deprecated the "dir" versions and add new `static_path`, `bokehjs_path` etc. The deprecated version could just call `str` on these.

Reply:
I did the later for `bokehjsdir()`, `settings.bokehjsdir()` and `serverdir()`.